apps/prairielearn/src/ee/lib/ai-instance-question-grouping/ai-instance-question-grouping.ts

```diff
@@ -112,48 +110,41 @@ async function aiEvaluateStudentResponse({
       content: answer_text,
     },
     {
-      role: 'developer',
+      role: 'system',
       content: 'Now, consider the following student submission:',
     },
     submissionMessage,
     {
-      role: 'developer',
+      role: 'system',
       content:
         'Please evaluate whether or not the student submission is correct according to the above instructions.',
     },
   ];
 
-  const response = await openai.responses.parse({
-    model: INSTANCE_QUESTION_GROUPING_OPENAI_MODEL,
-    input,
-    text: {
-      format: zodTextFormat(
-        z.object({
-          correct: z.boolean().describe('Whether or not the student submission is correct.'),
-        }),
-        'response-evaluation',
-      ),
+  const response = await generateObject({
+    model,
+    schema: z.object({
+      correct: z.boolean().describe('Whether or not the student submission is correct.'),
+    }),
+    messages: input,
+    providerOptions: {
+      openai: {
```

Comment:
This is the third time this is duplicated. Can we have a function like `buildMetadata`, and pass in the metadata to the function instead of passing all these ids to the function that are only used for this?

Reply:
Just because they're the same now doesn't mean they will be forever. I deduped this in the AI grading code in ce38e1d. I don't yet see the need to dedupe between AI grading and AI instance question grouping.